### ISSUE TITLE
Select _avg from relation and also return the parent entity

### ISSUE THREAD

**Issue by lgraubner** [prisma/prisma#10866]
## Problem

At the moment it's not possible to get a single entity from the database with prisma including an average of a relations field.

## Suggested solution

Allow `_avg` on relations, something like this:

```
let courses = await prisma.course.findMany({
  where: {
    topic: {
      slug: topic.slug,
    },
  },
  select: {
    id: true,
    name: true,
    slug: true,
    reviews: {
      _avg: {
        rating: true,
      },
    },
  },
});
```

## Alternatives

What works right now is getting the courses in one query and make a second query to get the averages. This has to be merged manually afterwards.

**Comment by alexitelman** [comment]
we need this too please

**Comment by lucasbriandbr** [comment]
please yes we need this, it will be so practical..

**Comment by ShaikRehan123** [comment]
Please so many need this.

**Comment by kevin1193** [comment]
yes would be better if we could implement this.

**Comment by danielimmke** [comment]
Yes this would be so good

**Comment by joshwcorbett** [comment]
any traction or updates on this?

**Comment by JClackett** [comment]
Yes please!! have the exact same setup, a review table with a rating field - having to break into raw sql (which is not so bad) to get average rating.

**Comment by amirbenasr** [comment]
YES PLEASE JUST DO IT ^^

**Comment by iyansr** [comment]
Any update?

**Comment by 1ron** [comment]
Anything?
